packages/caliper-core/test/worker/rate-control/fixedFeedbackRate.js

```diff
@@ -108,41 +108,176 @@ describe('fixedFeedbackRate controller implementation', () => {
             clock.restore();
         });
 
-        it('should not sleep if the generalSleepTime is 0', () => {
+        it('should not sleep if the generalSleepTime is 0', async () => {
             controller.generalSleepTime = 0;
             txnStats.stats.txCounters.totalSubmitted = 2000;
-            controller.applyRateControl();
+            await controller.applyRateControl();
 
             // should have not called the sleep method
             sinon.assert.notCalled(sleepStub);
         });
 
-        it('should not sleep if there are no unfinished transactions', () => {
-            txnStats.stats.txCounters.totalSubmitted = 0;
-            controller.applyRateControl();
+        it('should not sleep if there are no unfinished transactions', async () => {
+            // Stub methods if necessary
```

Comment:
Why stub only in this test when all others, including the original test just set the stats directly ?